engine/apps/mattermost/models/mattermost_channel.py

```diff
@@ -0,0 +1,38 @@
+from django.conf import settings
+from django.core.validators import MinLengthValidator
+from django.db import models
+
+from common.public_primary_keys import generate_public_primary_key, increase_public_primary_key_length
+
+
+def generate_public_primary_key_for_slack_channel():
+    prefix = "M"
+    new_public_primary_key = generate_public_primary_key(prefix)
+
+    failure_counter = 0
+    while MattermostChannel.objects.filter(public_primary_key=new_public_primary_key).exists():
+        new_public_primary_key = increase_public_primary_key_length(
+            failure_counter=failure_counter, prefix=prefix, model_name="MattermostChannel"
+        )
+        failure_counter += 1
+
+    return new_public_primary_key
+
+
+class MattermostChannel(models.Model):
+    organization = models.ForeignKey(
+        "user_management.Organization",
+        on_delete=models.CASCADE,
+        related_name="mattermost_channels",
+    )
+
+    public_primary_key = models.CharField(
+        max_length=20,
+        validators=[MinLengthValidator(settings.PUBLIC_PRIMARY_KEY_MIN_LENGTH + 1)],
+        unique=True,
+        default=generate_public_primary_key_for_slack_channel,
+    )
+
+    channel_id = models.CharField(max_length=100, default=None)
+    channel_name = models.CharField(max_length=100, default=None)
```

Comment:
Yup storing display_name can be helpful for displaying in UI as channel_name is the kebab case of the display_name added the same to the model

Reply:
I gave a thought about storing the team model but at this point I don't see much of use case for that, All post API are tied to channel or user not to team. The team info is mainly useful on some of the channel APIs for which i don't see a use case. Let me know if you thing otherwise.

Reply:
Ack, I was checking since it seems `team_name` is needed to register the channel, it may make sense to store something around team to make sure we can disambiguate later if needed? (at least `team_id`?)